python/lsst/pipe/analysis/colorAnalysis.py

```diff
@@ -363,281 +564,1057 @@ def plotGalaxyColors(self, catalogs, filenamer, dataId):
+            principalColorStr = makeEqnStr(principalColorStr, transform.coeffs.values(), wPerpFilters)
+            principalColorStrs.append(principalColorStr)
+
+        xLoc = xRange[1] - 0.03*deltaX
+        yLoc -= 0.05*deltaY
+        axes[0].text(xLoc, yLoc, wPerpStr, fontsize=6, ha="right", va="center", color="magenta")
+        yLoc -= 0.04*deltaY
+        axes[0].text(xLoc, yLoc, principalColorStrs[0], fontsize=6, ha="right", va="center",
+                     color="blue", alpha=0.8)
+        yLoc -= 0.05*deltaY
+        axes[0].text(xLoc, yLoc, wParaStr, fontsize=6, ha="right", va="center", color="magenta")
+        yLoc -= 0.04*deltaY
+        axes[0].text(xLoc, yLoc, principalColorStrs[1], fontsize=6, ha="right", va="center",
+                     color="blue", alpha=0.8)
+        log.info("{0:s}".format(wPerpStr))
+        log.info("{0:s}".format(wParaStr))
+
+        # Compute fitted P2 for each object
+        if transform is not None:
+            fitP2 = np.ones(numGood)*pColCoeffs.p2Coeffs[3]
+            for i, ff in enumerate(transform.coeffs.keys()):
+                if ff != "":
+                    fitP2 += mags[ff]*pColCoeffs.p2Coeffs[i]
+            fitP2 *= unitScale
 
     # Determine quality of locus
     distance2 = []
-    poly = np.poly1d(poly)
-    polyDeriv = np.polyder(poly)
-    calculateDistance2 = lambda x1, y1, x2: (x2 - x1)**2 + (poly(x2) - y1)**2
-    for x, y in zip(xx[select], yy[select]):
-        roots = np.roots(np.poly1d((1, -x)) + (poly - y)*polyDeriv)
-        distance2.append(min(calculateDistance2(x, y, np.real(rr)) for rr in roots if np.real(rr) == rr))
+    polyFit = np.poly1d(polyFit)
+    polyDeriv = np.polyder(polyFit)
+    for x, y in zip(xx[kept], yy[kept]):
+        roots = np.roots(np.poly1d((1, -x)) + (polyFit - y)*polyDeriv)
+        distance2.append(min(
+                distanceSquaredToPoly(x, y, np.real(rr), polyFit) for rr in roots if np.real(rr) == rr))
     distance = np.sqrt(distance2)
-    distance *= np.where(yy[select] >= poly(xx[select]), 1.0, -1.0)
-    unitStr = "mmag" if unitScale == 1000 else "mag"
+    distance *= np.where(yy[kept] >= polyFit(xx[kept]), 1.0, -1.0)
     distance *= unitScale
     q1, median, q3 = np.percentile(distance, [25, 50, 75])
     good = np.logical_not(np.abs(distance - median) > 3.0*0.74*(q3 - q1))
+    mean = distance[good].mean()
+    stdDev = distance[good].std()
+    rms = np.sqrt(np.mean(distance[good]**2))
     log.info(("Statistics from {0:} of Distance to polynomial ({9:s}): {7:s}\'star\': " +
               "Stats(mean={1:.4f}; stdev={2:.4f}; num={3:d}; total={4:d}; " +
               "median={5:.4f}; clip={6:.4f}; forcedMean=None){8:s}").format(
-            dataId, distance[good].mean(), distance[good].std(), len(xx[keep]), len(xx),
-            np.median(distance[good]), 3.0*0.74*(q3 - q1), "{", "}", unitStr))
+            dataId, mean, stdDev, len(xx[keep]), len(xx), np.median(distance[good]),
```

Comment:
both dataId and the "3" below should line up with the second `(`, 9 in from the log.